dot/core/service.go

```diff
@@ -372,17 +380,33 @@ func (s *Service) maintainTransactionPool(block *types.Block) {
 		s.transactionState.RemoveExtrinsic(ext)
 	}
 
+	bestBlockHash := s.blockState.BestBlockHash()
+	stateRoot, err := s.storageState.GetStateRootFromBlock(&bestBlockHash)
+	if err != nil {
+		logger.Errorf("could not get state root from block %s: %w", bestBlockHash, err)
```

Comment:
Shouldn't we return an error instead here?

Reply:
hmm this one maybe we should. Will ask for @timwu20 or @noot thoughts

Reply:
I agree, we return an error for `handleChainReorg` so I'm not sure why this function doesn't

Reply:
I don't think we should be continuing with the rest of this function if there is an error trying to get the `stateRoot`.